### PR TITLE
test: add integration tests for sidechain transactions

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

Title says it all.

### Context of Change

Integration tests for #2039. Any bugfixes that were caught by these tests were moved to #2039.

### Type of Change

- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

## Test Plan

CI passes.

### CODE REVIEW

**Comment by mvadari** [comment]
Note: integration tests pass locally when I run them, but the Docker container hasn't been updated yet

**Comment by intelliot** [comment]
note: this needs review and is blocking #2039

**Comment by mvadari** [comment]
> note: this needs review and is blocking #2039

It's fine if this PR has to wait for the Docker container to be updated.

File: packages/ripple-binary-codec/src/types/xchain-bridge.ts [hunk 1]
```diff
@@ -67,20 +67,20 @@ class XChainBridge extends SerializedType {
...
-        const object = type.from(value[name])
-        bytes.push(object.toBytes())
-      })
-      return new XChainBridge(Buffer.concat(bytes))
+    if (!isXChainBridgeObject) {
+      throw new Error('Invalid type to construct an XChainBridge')
     }
 
-    throw new Error('Invalid type to construct an XChainBridge')
+    const bytes: Array<Buffer> = []
```
Comment (ckniffen) [inline packages/ripple-binary-codec/src/types/xchain-bridge.ts:74]: Big fan of this flop.

File: packages/xrpl/test/integration/transactions/xchainClaim.test.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    'base',
+    async () => {
+      const signatureReward = '1000000'
+      const xchainBridge: XChainBridge = {
+        LockingChainDoor: testContext.wallet.classicAddress,
+        LockingChainIssue: { currency: 'XRP' },
+        IssuingChainDoor: GENESIS_ACCOUNT,
+        IssuingChainIssue: { currency: 'XRP' },
+      }
+      const setupTx: XChainCreateBridge = {
```
Comment (ckniffen) [inline packages/xrpl/test/integration/transactions/xchainClaim.test.ts:50]: Could this setup code be moved to a utility method?  Similar to what was done for the AMM integration tests.

**Review by ckniffen — APPROVED** [review]

**Review by jonathanlei — APPROVED** [review]

**Pull request merged**